### PR TITLE
tests/compile/compile.sh improvements

### LINKED ISSUE #1873: tests/compile/compile.sh: Adaptively find the correct path
Adaptively find the correct path. Allow the script to run from the local directory (and possibly anywhere within the Blockly project).

Don't depend on `npm root` to find the path. (For instance, if someone is using a local `*compiler*.jar`.)

### PR DESCRIPTION
**Pull request by AnmAtAnm**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

#1873 and numerous other minor issues.

### Proposed Changes

 * Allow running from either Blockly root or local dir.
 * Allow the use of npm google-closure-library.
 * Log compiler/library versions
 * Log full *compiler*.jar filename. Limit to one match.
 * Add npm instructions for both compiler and library.
 * Minor stylistic polish.

### Reason for Changes

Improvements since I'm familiar with the code from previous PR.

### Test Coverage

 * I built from blockly root and from `tests/compile`.
 * I built with `npm` packages installed, and with local compiler `.jar` downloaded (from both locations)
 * Verified the output reflected the expected configuration and passed.
 * I ran with multiple `*compiler*.jar` files and watched it fail, listing all matches found.

### CODE REVIEW

**Comment by AnmAtAnm** [comment]
Beginning of output: https://travis-ci.org/google/blockly/jobs/379968416#L1736

File: tests/compile/compile.sh [hunk 1]
```diff
@@ -1,42 +1,99 @@
...
-  echo "Download from this URL, and place jar file in current directory."
-  echo "https://dl.google.com/closure-compiler/compiler-latest.zip"
+  COMPILER_JARS=$(find $BLOCKLY_ROOT/tests/compile/ -maxdepth 1 -name "*compiler*.jar")
+  if [ -n "$COMPILER_JARS" ]; then
+    if [ $(echo "$COMPILER_JARS" | wc -l) -ne 1 ]; then
+      echo "ERROR: Found too many Closure *compiler*.jar files:" 1>&2;
+      echo "$COMPILER_JARS" 1>&2;
+      exit 1
+    fi
+    # TODO: Check whether multiple files were found.
```
Comment (rachel-fenichel) [inline tests/compile/compile.sh]: This todo is done now.

File: tests/compile/compile.sh [hunk 2]
```diff
@@ -1,42 +1,99 @@
...
+  BLOCKLY_ROOT="../.."
+elif [ -f tests/compile/compile.sh ]; then
+  BLOCKLY_ROOT="."
+else
+  echo "ERROR: Cannot determine BLOCKLY_ROOT" 1>&2;
+  exit 1
+fi
+
+# Test for npm and node_modules directory.
+if command -v npm >/dev/null 2>&1; then
```
Comment (rachel-fenichel) [inline tests/compile/compile.sh]: What does `command -v npm` do?

**Comment by rachel-fenichel** [comment]
Also, a low-priority follow-up: we may want to force log folding, since this is currently very verbose: https://github.com/travis-ci/travis-ci/issues/2285

**Comment by AnmAtAnm** [comment]
I'm going to attempt to address log folding in the next PR at the `.travis.yml` level.

**Pull request merged**